spec/acceptance/otelcol_contrib_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'Otelcol class' do
+  context 'otelcol-contib', 'init.pp' do
+    # Using puppet_apply as a helper
+    # it_behaves_like 'init', 'init.pp'
+    it 'works idempotently with no errors' do
+      pp = <<-EOS
+      class { 'otelcol':
+        manage_archive    => true,
+        package_name      => 'otelcol-contrib',
+        config_file_owner => 'otelcol-contrib',
+        config_file_group => 'otelcol-contrib',
+        metrics_address_port => 8889,
+        receivers         => {
+          'otlp'       => {
+            'protocols' => {
+              'grpc' => { 'endpoint' => 'localhost:4319' },
+              'http' => { 'endpoint' => 'localhost:4320' },
+            },
+          },
+          'prometheus' => {
+            'config' => {
+              'scrape_configs' => [
+                {
+                  'job_name'        => 'otel-collector',
+                  'scrape_interval' => '10s',
+                  'static_configs'  => [
+                    { 'targets' => ['localhost:8889'] }
+                  ],
+                },
+              ],
+            },
+          },
+        },
+        exporters         => { 'logging' => { 'verbosity' => 'detailed' } },
+        pipelines         => {
+          'metrics' => {
+            'receivers' => ['otlp', 'prometheus'],
+            'exporters' => ['logging'],
+          },
+        },
+        processors        => { 'batch' => {} },
+      }
+      EOS
+      # Run it twice and test for idempotency
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
```

Comment:
This really require 3 runs?  Otherwise we have an handy helper:
https://github.com/voxpupuli/voxpupuli-acceptance#an-idempotent-resource

Reply:
I've changed that